pallets/keystore/src/lib.rs

```diff
@@ -56,9 +71,23 @@ pub mod pallet {
             Ok(())
         }
 
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn announce_key(
+            origin: OriginFor<T>,
+            fingerprint: Vec<u8>,
+            location: Vec<u8>,
+        ) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            <IssuedKeys<T>>::insert(&who, &fingerprint, &location);
+
+            Self::deposit_event(Event::KeyExists(fingerprint, location, who));
+            Ok(())
+        }
+
         /// If a key needs to be removed from circulation, this extrinsic will handle deleting it
         /// and informing the network.
-        #[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+        #[pallet::weight(26_623_000 + T::DbWeight::get().reads_writes(1,1))]
```

Comment:
You can call the weights.rs functions directly in these pallet::weight proc macros. That way if you update weights, you can just copy over to weights.rs and don't have to do any line-swaps.

Reply:
@Romulus10 Should I be able to just copy the generated file to weights.rs? I see the output doesn't quite look like what's in identity and trust. I think I'll push up my latest generated files shortly.

Reply:
You should, yeah. I'll take a look at it once you push.

Reply:
@Romulus10 Latest commit is weights.rs exactly as generated.